### PR TITLE
added test for map save

### LINKED ISSUE #2303: Map save needs a test
The save method for Map needs a test.  It doesn't have one.  I suggest using the hash approach we use for testing our plotting code.  What do you think?  The same approach will also be useful in creating a save test for mapcube - see #2301.

### PR DESCRIPTION
**Pull request by himanshukgp**
Closes #2303

### CODE REVIEW

File: sunpy/map/tests/test_mapbase.py [hunk 1]
```diff
@@ -280,6 +288,15 @@ def test_world_to_pixel(generic_map):
     assert_quantity_allclose(test_pixel, generic_map.reference_pixel)
 
 
+def test_save(generic_map):
+        # Test save out
+        eitmap = sunpy.map.Map(a_fname)
+        afilename = tempfile.NamedTemporaryFile(suffix='fits').name
+        eitmap.save(afilename, filetype='fits', clobber=True)
+        backin = sunpy.map.Map(afilename)
+        assert isinstance(backin, sunpy.map.sources.EITMap)
```
Comment (Cadair) [inline sunpy/map/tests/test_mapbase.py]: adding some test to make sure the `data` and `meta` attributes survied the trip would be very valuable.

**Comment by Cadair** [comment]
Thanks @himanshukgp

File: sunpy/map/tests/test_mapbase.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/map/tests/test_mapbase.py]: Like this?

**Comment by nabobalis** [comment]
I just added two extra asserts for the data and header.

**Comment by nabobalis** [comment]
@himanshukgp Thanks for this! Sorry for getting a bits handsy on it, I was just thinking of adding this for our next release (0.8.3)!

**Pull request merged**